### PR TITLE
Redesign base path file structure

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
Addresses #368

First review this #372

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
Thanks @PhilippeMoussalli!

Is the manifest now also saved in the same directory as the data when running on Kubeflow? Or is it only written as a Kubeflow artifact?

**Comment by PhilippeMoussalli** [comment]
> Thanks @PhilippeMoussalli!
> 
> Is the manifest now also saved in the same directory as the data when running on Kubeflow? Or is it only written as a Kubeflow artifact?

In both actually, It was done in #322 

The kubeflow artifact one (which can't be modified) is needed to connect components through artifacts and the data one (in the base path) is for us

File: src/fondant/executor.py [hunk 1]
```diff
@@ -212,9 +212,9 @@ def upload_manifest(self, manifest: Manifest, save_path: t.Union[str, Path]):
 
         if is_kubeflow_output:
```
Comment (RobbeSneyders) [inline src/fondant/executor.py:213]: I'm not the biggest fan of having kubeflow specific code in here. Opened #376 to remove this.

**Review by RobbeSneyders — COMMENTED** [review]
Ok thanks, that clears things up.

File: src/fondant/executor.py [hunk 2]
```diff
@@ -212,9 +212,9 @@ def upload_manifest(self, manifest: Manifest, save_path: t.Union[str, Path]):
...
             # Save to the expected base path directory
-            safe_component_name = self.spec.name.replace(" ", "_").lower()
             save_path_base_path = (
-                f"{manifest.base_path}/{safe_component_name}/manifest.json"
+                f"{manifest.pipeline_name}/{manifest.run_id}/"
+                f"{manifest.component_id}/manifest.json"
```
[the remark this answers is not in the document]
Comment (PhilippeMoussalli) [reply src/fondant/executor.py:217]: Good catch! I confused it with a subset where the `base_path` is appended to the location

File: src/fondant/executor.py (hunk 1 above)
Comment (PhilippeMoussalli) [reply src/fondant/executor.py:213]: I agree, though I don't see a clear-cut solution for this just yet

Comment (GeorgesLorre) [reply src/fondant/executor.py:213]: will vertex have the same issue ?

Comment (PhilippeMoussalli) [reply src/fondant/executor.py:213]: probably, ideally we have some control over the format and save path of the artifact.

**Review by RobbeSneyders — APPROVED** [review]
Thanks @PhilippeMoussalli!

**Pull request merged**